models/inhomogeneous_poisson_generator.cpp

```diff
@@ -180,17 +179,16 @@ nest::inh_poisson_generator::update( Time const& origin,
 
     // Keep the amplitude up-to-date at all times.
     // We need to change the amplitude one step ahead of time, see comment
-    // on class SimulatingDevice.
-    if ( B_.idx_ < P_.rate_times_.size()
-      && curr_time + 1
-        == Time( Time::ms( P_.rate_times_[ B_.idx_ ] ) ).get_steps() )
+    // on class StimulatingDevice.
+    if ( B_.idx_ < P_.rate_times_.size() && curr_time + 1
+      == Time( Time::ms( P_.rate_times_[ B_.idx_ ] ) ).get_steps() )
```

Comment:
This can go wrong if subsequent values in `rate_times_` are closer to each other than resolution; see #740 for a discussion of the problem. As discussed there, we probably should convert the times to `Time` objects via `Time::ms_stamp()`, so rates can only change at time steps.